content/src/thumbnail.js

```diff
@@ -1,20 +1,50 @@
+var html2canvas = require('html2canvas');
 var THUMBNAIL_SIZE = 128;
+var NUM_ATTEMPTS = 3;
 
 // Public functions
 var thumbnail = {
-  generateThumbnailDataUrl: function(iframe) {
+  generateThumbnailDataUrl: function(iframe, callback) {
     // Get the canvas inside the iframe.
     var innerDoc = iframe.contentDocument || iframe.contentWindow.document;
-    // TODO: Add support for multiple turtle.
-    var canvas = innerDoc.getElementsByTagName('canvas')[0];
+    var innerBody = innerDoc.body;
 
-    if (!canvas || canvas.id === 'turtle') { return; }
+    // Hide the test panel and coordinates before capturing the thumbnail.
+    for (var i = 0; i < innerBody.childElementCount; i++) {
+      if (innerBody.children[i].tagName.toLowerCase() === 'samp' && (
+          innerBody.children[i].className !== 'turtlefield' ||
+          innerBody.children[i].id == '_testpanel')) {
+        innerBody.children[i].style.display = 'none';
```

Comment:
Here maybe have a list of `hiddenElements`, and then accumulate the list as well as the old value of `elt.style.display` when you set each `elt.style.display = 'none'`.

Then below in `onRendered`, you can go through the same list and restore all the old `elt.style.display` values.